File: 01_GOF23/DP_examples/Python/03_Behavioral/Memento/caretaker.py
# 管理者类
class Caretaker:
    def __init__(self):
        self.memento_map = {}
    
    def save_memento(self, version, m):
        self.memento_map[version] = m
    
    def get_memento(self, version):
        return self.memento_map.get(version)

File: 01_GOF23/DP_examples/Lua/03_Behavioral/Memento/Caretaker.lua
-- 管理者类
local Caretaker = {}
Caretaker.__index = Caretaker

function Caretaker:new()
    local o = {}
    setmetatable(o, self)
    o.mementoMap = {}
    return o
end

function Caretaker:SaveMemento(version, m)
    self.mementoMap[version] = m
end

function Caretaker:GetMemento(version)
    return self.mementoMap[version]
end

return Caretaker
